adr/component_offerings.md

```diff
@@ -0,0 +1,134 @@
+# Offer both low-level and high-level components that are exported from each package.
+
+## Context
+
+The primary issue we are facing right now is determining how "high-level" the component should be. For example, should we create a single component with many props, or should we create many components to be composed together.
+
+This boils down to two main goals:
+
+-   Provide enough flexibility and low-level components for teams to quickly build and deliver applications without being blocked by us
+-   Provide quality high level components that allow teams to quickly build applications with the best possible interactions (UX) that are consistent, performant, accessible and extensible without the need of repeating themselves over and over again.
+
+```jsx
+<ClayCard>
+	<ClayCard.Title onClick={handleTitleClick}>
+		<ClayIcon symbol="check" />
+
+		{'Hello World'}
+	</ClayCard.Title>
+
+	<ClayCard.Body onClick={handleBodyClick}>
+		<ClayCard.Description>
+			{'Some cool descriptors'}
+		</ClayCard.Description>
+	</ClayCard.Body>
+</ClayCard>
+
+// vs
+<ClayCard
+	iconTitleLeftSymbol="check"
+	onTitleClick={handleTitleClick}
+	onBodyClick={handleBodyClick}
+	description="foo bar baz"
+	title="hello world"
+/>
+```
+
+## Decision
+
+The decision going forward is to create 2 sets of components
+
+-   low-level composable components such as `<ClayCard.Header>`, `<ClayCard.Footer>`
+-   simple high-level components that cover basic lexicon uses
+
+### What does this look like?
+
+#### Composable:
+
+![Composable Drop Down](./images/component_offerings_2.png)
+
+```jsx
+<ClayDropDown>
+	<ClayDropDown.Search
+		onChange={() => {}}
```

Comment:
In this example I'm assming that the `onChange` and `value` attributes are optional? 
(Might be totally off topic but I thought I'd ask)

Reply:
These would be required props to make it functional. This gives the user flexibility to do whatever they need when the input value changes.

However, in a high-level component, this would be handled internally.

Reply:
@bryceosterhaus 👍 Thanks for making that clear